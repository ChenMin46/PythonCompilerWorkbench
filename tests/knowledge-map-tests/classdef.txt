class Person:
    pass

class Person(object):
    def grow_older(self, x):
        self.age += x

    def __str__(self):
        return 'Person: ' + self.name


class Man(Person):
    maximum_age = 100
