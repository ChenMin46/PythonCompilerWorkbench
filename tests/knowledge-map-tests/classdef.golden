class Person:
    pass

class Person(object):
    def grow_older(self, x):
        self.age += x

    def __str__(self):
        return 'Person: ' + \
        \
\
        self.name


class Man(Person):
    maximum_age = 100
===
Module
  body: [
    ClassDef >> 'class Person'
      name: 'Person'
      bases: [
      ]
      body: [
        Pass >> 'pass'
      ]
      decorator_list: [
      ]
    ClassDef >> 'class Person(object)'
      name: 'Person'
      bases: [
        Name >> 'object'
          id: 'object'
          ctx: Load
      ]
      body: [
        FunctionDef >> 'def grow_older(self, x)'
          name: 'grow_older'
          args: arguments
            args: [
              Name >> 'self'
                id: 'self'
                ctx: Param
              Name >> 'x'
                id: 'x'
                ctx: Param
            ]
            vararg: None
            kwarg: None
            defaults: [
            ]
          body: [
            AugAssign >> 'self.age += x'
              target: Attribute >> 'self.age'
                value: Name >> 'self'
                  id: 'self'
                  ctx: Load
                attr: 'age'
                ctx: Store
              op: Add
              value: Name >> 'x'
                id: 'x'
                ctx: Load
          ]
          decorator_list: [
          ]
        FunctionDef >> 'def __str__(self)'
          name: '__str__'
          args: arguments
            args: [
              Name >> 'self'
                id: 'self'
                ctx: Param
            ]
            vararg: None
            kwarg: None
            defaults: [
            ]
          body: [
            Return ML>> "return 'Person: ' + \\\n        \\\n\\\n        self.name"
              value: BinOp ML>> "'Person: ' + \\\n        \\\n\\\n        self.name"
                left: Str >> "'Person: '"
                  s: 'Person: '
                op: Add
                right: Attribute >> 'self.name'
                  value: Name >> 'self'
                    id: 'self'
                    ctx: Load
                  attr: 'name'
                  ctx: Load
          ]
          decorator_list: [
          ]
      ]
      decorator_list: [
      ]
    ClassDef >> 'class Man(Person)'
      name: 'Man'
      bases: [
        Name >> 'Person'
          id: 'Person'
          ctx: Load
      ]
      body: [
        Assign >> 'maximum_age = 100'
          targets: [
            Name >> 'maximum_age'
              id: 'maximum_age'
              ctx: Store
          ]
          value: Num >> '100'
            n: 100
      ]
      decorator_list: [
      ]
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "id": "id_0", 
          "name": "ClassDef", 
          "value": "class Person"
        }, 
        {
          "id": "tid_1", 
          "name": "token", 
          "value": ":"
        }, 
        "\n    ", 
        {
          "contents": [
            {
              "id": "tid_2", 
              "name": "token", 
              "value": "pass"
            }
          ], 
          "id": "id_0", 
          "name": "Pass"
        }, 
        "\n\n", 
        {
          "contents": [
            "class Person(", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "object"
            }, 
            {
              "id": "tid_3", 
              "name": "token", 
              "value": ")"
            }
          ], 
          "id": "id_0", 
          "name": "ClassDef"
        }, 
        {
          "id": "tid_4", 
          "name": "token", 
          "value": ":"
        }, 
        "\n    ", 
        {
          "contents": [
            "def grow_older(", 
            {
              "contents": [
                {
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "self"
                }, 
                {
                  "id": "tid_5", 
                  "name": "token", 
                  "value": ","
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "x"
                }
              ], 
              "id": "id_0", 
              "name": "arguments"
            }, 
            {
              "id": "tid_6", 
              "name": "token", 
              "value": ")"
            }
          ], 
          "id": "id_0", 
          "name": "FunctionDef"
        }, 
        {
          "id": "tid_7", 
          "name": "token", 
          "value": ":"
        }, 
        "\n        ", 
        {
          "contents": [
            {
              "contents": [
                {
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "self"
                }, 
                ".age"
              ], 
              "id": "id_0", 
              "name": "Attribute"
            }, 
            " ", 
            {
              "id": "tid_8", 
              "name": "token", 
              "value": "+="
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }
          ], 
          "id": "id_0", 
          "name": "AugAssign"
        }, 
        "\n\n    ", 
        {
          "contents": [
            "def __str__(", 
            {
              "contents": [
                {
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "self"
                }
              ], 
              "id": "id_0", 
              "name": "arguments"
            }, 
            {
              "id": "tid_9", 
              "name": "token", 
              "value": ")"
            }
          ], 
          "id": "id_0", 
          "name": "FunctionDef"
        }, 
        {
          "id": "tid_10", 
          "name": "token", 
          "value": ":"
        }, 
        "\n        ", 
        {
          "contents": [
            {
              "id": "tid_11", 
              "name": "token", 
              "value": "return"
            }, 
            " ", 
            {
              "contents": [
                {
                  "id": "id_0", 
                  "name": "Str", 
                  "value": "'Person: '"
                }, 
                " + \\\n        ", 
                {
                  "id": "tid_12", 
                  "name": "token", 
                  "value": "\\"
                }, 
                "\n", 
                {
                  "id": "tid_13", 
                  "name": "token", 
                  "value": "\\"
                }, 
                "\n        ", 
                {
                  "contents": [
                    {
                      "id": "id_0", 
                      "name": "Name", 
                      "value": "self"
                    }, 
                    ".name"
                  ], 
                  "id": "id_0", 
                  "name": "Attribute"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Return"
        }, 
        "\n\n\n", 
        {
          "contents": [
            "class Man(", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "Person"
            }, 
            {
              "id": "tid_14", 
              "name": "token", 
              "value": ")"
            }
          ], 
          "id": "id_0", 
          "name": "ClassDef"
        }, 
        {
          "id": "tid_15", 
          "name": "token", 
          "value": ":"
        }, 
        "\n    ", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "maximum_age"
            }, 
            " ", 
            {
              "id": "tid_16", 
              "name": "token", 
              "value": "="
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "100"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
